streams/src/main/java/org/apache/kafka/streams/state/internals/WrappedStateStore.java

```diff
@@ -93,5 +97,15 @@ public void close() {
         public StateStore wrappedStore() {
             return innerState;
         }
+
+        void throwException(final KafkaException exception, final String message) {
+            if (exception instanceof ProducerFencedException) {
+                throw new ProducerFencedException(message);
+            } else if (exception instanceof ProcessorStateException) {
+                throw new ProcessorStateException(message, exception);
+            } else {
+                throw new StreamsException(message, exception);
+            }
+        }
```

Comment:
It might be a personal preference, but I'd rather just suffer the duplication of having separate catch blocks for separate kinds of exceptions.